proj/src/main/webapp/index.html

```diff
@@ -11,13 +11,26 @@
         <!-- TODO: Add SDKs for Firebase products that you want to use
             https://firebase.google.com/docs/web/setup#available-libraries -->
         <script src="https://www.gstatic.com/firebasejs/7.15.4/firebase-analytics.js"></script>
+
+        <script src="app.js"></script>
+
+        <script src="https://www.gstatic.com/firebasejs/7.15.4/firebase-auth.js"></script>
         <script src="https://www.gstatic.com/firebasejs/7.15.4/firebase-database.js"></script>
         <script src="https://www.gstatic.com/firebasejs/7.15.4/firebase-messaging.js"></script>
 
-        <script src="app.js"></script>
-        <script src="script.js"></script>
     </head>
     <body onload="init()">
+        <div class="login-container">
+            <input id="email" type="email" placeholder="Email">
+            <input id="pass" type="password" placeholder="Password">
+            
+            <button id="btnLogin">Log in</button>
+            <button id="btnSignUp">Sign Up</button>
+            <button id="btnLogout" class="hidden">Log out</button>
+        </div>
+
+        <script src="script.js"></script>
```

Comment:
Why isn't this script with all the other scripts above?

Reply:
this is also my question

Reply:
This script needs to be after the authentication objects in order for it to run properly. less than Ideal, but this works until we discover a workaround.

Reply:
Is it so that the const variables at the top can be initialized properly?  Can that be fixed by creating them in an init function and calling that when the page is done loading?  I think that would be better in general...although I don't really know what the best practices for JavaScript are.

Reply:
Script tags are normally in the head or bottom of the body (so the script runs after the page is pared and loaded). There are also the async and defer options to choose when to load the script.